pkg/volume/csi/csi_plugin.go

```diff
@@ -246,16 +248,84 @@ func (p *csiPlugin) Init(host volume.VolumeHost) error {
 		}
 	}
 
-	// Initializing csiDrivers map and label management channels
+	// Initializing csiDrivers map, label management channels, and migratedDrivers map
 	csiDrivers = csiDriversStore{driversMap: map[string]csiDriver{}}
-	nim = nodeinfomanager.NewNodeInfoManager(host.GetNodeName(), host)
 
-	// TODO(#70514) Init CSINodeInfo object if the CRD exists and create Driver
-	// objects for migrated drivers.
+	var migratedDrivers = map[string](func() bool){
+		csiPlugins.GCEPDDriverName: func() bool {
+			return utilfeature.DefaultFeatureGate.Enabled(features.CSIMigration) && utilfeature.DefaultFeatureGate.Enabled(features.CSIMigrationGCE)
+		},
+		// TODO(leakingtpan): Add AWS migration feature gates and place them here
+	}
+
+	nim = nodeinfomanager.NewNodeInfoManager(host.GetNodeName(), host, migratedDrivers)
```

Comment:
should this only be initialized if CSINodeInfo is enabled?

assigning to a global here is a code smell. is there a reason not to put these in the csiPlugin instance?

Reply:
1. probably yes, I will gate it.

2. Not sure, I didn't actually write this code I just added the `migratedDrivers` parameter. Any bigger changes are outside the scope of this PR as I'm not sure what side effects they may have
/cc @verult 
who I believe is familiar with (or wrote) this code

Reply:
@mariantalla and /me are planning to address the globals in `pkg/volume/csi/...`, so I suggest to leave it as is for now.